### PR TITLE
Fix onAnnotationDragStarted event is still fired when annotation is not draggable - MAPSAND-356

### PR DESCRIPTION
**Pull request by Chaoba**
### Summary of changes
AnnotationManager fires `onAnnotationDragStarted` events even the annotation is not draggable.  In this pr, we will check the `isDraggable` property when start dragging and return directly if the annotation is not draggable so that `onAnnotationDragStarted` event will not be fired.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

**Review by ank27 — APPROVED** [review]

File: plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt [hunk 1]
```diff
@@ -714,6 +714,9 @@ abstract class AnnotationManagerImpl<G : Geometry, T : Annotation<G>, S : Annota
     }
 
     private fun startDragging(annotation: T): Boolean {
+      if (!annotation.isDraggable) {
```
Comment (yunikkk) [inline plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt:717]: @Chaoba the only place where `startDragging` is called from is `onMoveBegin`, which also runs `queryMapForFeatures` before. Could we check `isDraggable` before and skip the query render features call then? Will save couple thread switches

Comment (yunikkk) [reply plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt:717]: E.g. [here ](https://github.com/mapbox/mapbox-maps-android/blob/4a633d6c118424ffa852dd6fde158ab48ddbb10b/plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt#L647)

Comment (Chaoba) [reply plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt:717]: Only after `queryMapForFeatures` can we get the annotation and know whether it's draggable or not.

Comment (yunikkk) [reply plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt:717]: Oh, indeed, somehow missed the obvious part

**Review by yunikkk — APPROVED** [review]

**Pull request merged**